Duplicati/Library/Backend/TencentCOS/COSBackend.cs

```diff
@@ -0,0 +1,431 @@
+                DeleteObjectRequest request = new DeleteObjectRequest(bucket, key);
+                //设置签名有效时长
+                request.SetSign(TimeUtils.GetCurrentTime(TimeUnit.SECONDS), 600);
+                //执行请求
+                DeleteObjectResult result = _cosXml.DeleteObject(request);
+                //请求成功
+                Console.WriteLine(result.GetResultInfo());
+            }
+            catch (COSXML.CosException.CosClientException clientEx)
+            {
+                //请求失败
+                Console.WriteLine("CosClientException: " + clientEx);
+                throw;
+            }
+            catch (COSXML.CosException.CosServerException serverEx)
+            {
+                //请求失败
+                Console.WriteLine("CosServerException: " + serverEx.GetInfo());
+                throw;
+            }
+        }
+
+        public void Test()
+        {
+            var json = JsonConvert.SerializeObject(_cosOptions);
+            try
+            {
+                _cosXml = GetCosXml();
+                string bucket = _cosOptions.Bucket;
+                HeadBucketRequest request = new HeadBucketRequest(bucket);
+                request.SetSign(TimeUtils.GetCurrentTime(TimeUnit.SECONDS), 600);
+                HeadBucketResult result = _cosXml.HeadBucket(request);
+
+                Logging.Log.WriteInformationMessage(LOGTAG, "Test", "Request complete {0}: {1}, {2}", result.httpCode, json, result.GetResultInfo());
+            }
+            catch (COSXML.CosException.CosClientException clientEx)
+            {
+                Logging.Log.WriteErrorMessage(LOGTAG, "Test", clientEx, "Request failed: {0}", json);
+                throw;
+            }
+            catch (COSXML.CosException.CosServerException serverEx)
+            {
+                Logging.Log.WriteErrorMessage(LOGTAG, "Test", serverEx, "Request failed: {0}, {1}", json, serverEx.GetInfo());
+                throw;
+            }
+        }
+
+        public void CreateFolder()
+        {
+
+        }
+
+        public void Dispose()
+        {
+            if (_cosXml != null)
+                _cosXml = null;
+        }
+
+        public Task PutAsync(string remotename, Stream stream, CancellationToken cancelToken)
+        {
+            try
+            {
+                _cosXml = GetCosXml();
+                string bucket = _cosOptions.Bucket;
+                string key = GetFullKey(remotename);
+
+                byte[] bytes = new byte[stream.Length];
+                stream.Read(bytes, 0, bytes.Length);
+                stream.Seek(0, SeekOrigin.Begin);
+
+                PutObjectRequest request = new PutObjectRequest(bucket, key, bytes);
+                //设置签名有效时长
+                request.SetSign(TimeUtils.GetCurrentTime(TimeUnit.SECONDS), KEY_DURATION_SECOND);
+                //执行请求
+                PutObjectResult result = _cosXml.PutObject(request);
+                //对象的 eTag
+                string eTag = result.eTag;
```

Comment:
This variable appears to be unused.  Did you intend to reference it somewhere?  If not, we should remove it.